sys/net/application_layer/ng_zep/ng_zep.c

```diff
@@ -0,0 +1,1079 @@
+        DEBUG("zep: dev was NULL\n");
+        return -ENODEV;
+    }
+
+    if ((dst == NULL) || (ng_ipv6_addr_is_unspecified(dst))) {
+        DEBUG("zep: dst (%s) was NULL or unspecified\n", dst);
+        return -ENOTSUP;
+    }
+
+    if (ng_netreg_lookup(NG_NETTYPE_UDP, src_port)) {
+        DEBUG("zep: port (%" PRIu16 ") already registered\n", src_port);
+        return -EADDRINUSE;
+    }
+
+    dev->driver = (ng_netdev_driver_t *)&_zep_driver;
+    dev->chan = NG_ZEP_DEFAULT_CHANNEL;
+    dev->pan = byteorder_btols(byteorder_htons(NG_ZEP_DEFAULT_PANID));
+    dev->flags = NG_ZEP_FLAGS_USE_SRC_PAN;
+#if CPUID_ID_LEN
+    /* initialize dev->addr and dev->eui64 from cpuid if available */
+    cpuid_get(cpuid);
+
+    hash1 = djb2_hash(cpuid, CPUID_ID_LEN / 2);
+    dev->addr.u16 = (uint16_t)((hash1 >> 16) ^ (hash1 & 0xffff));
+
+    if (CPUID_ID_LEN % 2) {
+        hash2 = djb2_hash(cpuid + (CPUID_ID_LEN / 2), (CPUID_ID_LEN / 2) - 1);
+    }
+    else {
+        hash2 = djb2_hash(cpuid + (CPUID_ID_LEN / 2), CPUID_ID_LEN / 2);
+    }
+
+    dev->eui64.u32[0] = hash1;
+    dev->eui64.u32[1] = hash2;
+
+    dev->eui64.u8[7] &= 0xfe;   /* set to unicast */
+    dev->eui64.u8[7] |= 0x02;   /* set to locally administered */
+#else
+    dev->addr = NG_ZEP_DEFAULT_ADDR_SHORT;
+    dev->eui64 = NG_ZEP_DEFAULT_ADDR_LONG;
+#endif
+    DEBUG("zep: initialized radio parameters: chan: %" PRIu8 ", pan: 0x%04" PRIx16
+          "addr: 0x%04" PRIx16 ", eui64: %016" PRIx64 "\n", dev->chan,
+          dev->pan, byteorder_ltobs(dev->addr).u16,
+          byteorder_ltobll(dev->eui64).u64);
+
+#ifdef MODULE_NG_SIXLOWPAN
+    dev->proto = NG_NETTYPE_SIXLOWPAN;
+#else
+    dev->proto = NG_NETTYPE_UNDEF;
+#endif
+
+    dev->seq = genrand_uint32();
+    dev->src_port = src_port;
+    dev->dst.u64[0] = dst->u64[0];
+    dev->dst.u64[1] = dst->u64[1];
+    dev->dst_port = dst_port;
+    dev->version = 2;
+    dev->lqi_mode = 1;
+
+    _pid = thread_create(_rx_stack, NG_ZEP_STACK_SIZE, NG_ZEP_PRIO,
+                         CREATE_STACKTEST, _event_loop, dev, "zep_app");
+
+    DEBUG("zep: started thread with PID %" PRIkernel_pid "\n", _pid);
+
+    return _pid;
+}
+
+/* helper functions for options to avoid type pruning */
+static inline void _set_uint16_ptr(uint16_t *ptr, uint16_t val)
+{
+    *ptr = val;
+}
+
+static inline void _set_uint64_ptr(uint64_t *ptr, uint64_t val)
+{
+    *ptr = val;
+}
+
+static inline void _set_flag_ptr(ng_netconf_enable_t *enable,
+                                 uint16_t flag_field, uint16_t flag)
+{
+    if (flag_field & flag) {
+        *enable = NETCONF_ENABLE;
+    }
+    else {
+        *enable = NETCONF_DISABLE;
+    }
+}
+
+static inline uint16_t *_get_uint16_ptr(void *ptr)
+{
+    return ptr;
+}
+
+static inline uint64_t *_get_uint64_ptr(void *ptr)
+{
+    return ptr;
+}
+
+static int _send(ng_netdev_t *netdev, ng_pktsnip_t *pkt)
+{
+    ng_zep_t *dev = (ng_zep_t *)netdev;
+    ng_pktsnip_t *ptr, *new, *hdr;
```

Comment:
From our coding conventions:

>     LOCAL variable names should be short, and to the point.  If you have
>     some random integer loop counter, it should probably be called "i".
>     Calling it "loop_counter" is non-productive, if there is no chance of it
>     being mis-understood.  Similarly, "tmp" can be just about any type of
>     variable that is used to hold a temporary value.

I argue, that `new` is to the point, since it is the newly generated packet.